micro-rdk/src/common/credentials_storage.rs

```diff
@@ -174,42 +195,76 @@ impl RAMStorage {
 
 #[cfg(feature = "ota")]
 impl OtaMetadataStorage for RAMStorage {
-    type Error = Infallible;
+    type Error = RAMStorageError;
     fn has_ota_metadata(&self) -> bool {
         let inner_ref = self.0.lock().unwrap();
         inner_ref.ota_metadata.is_some()
     }
-    fn store_ota_metadata(&self, ota_metadata: OtaMetadata) -> Result<(), Self::Error> {
+    fn store_ota_metadata(&self, ota_metadata: &OtaMetadata) -> Result<(), Self::Error> {
         let mut inner_ref = self.0.lock().unwrap();
-        let _ = inner_ref.ota_metadata.insert(ota_metadata);
+        let _ = inner_ref.ota_metadata.insert(ota_metadata.clone());
         Ok(())
     }
     fn get_ota_metadata(&self) -> Result<OtaMetadata, Self::Error> {
         let inner_ref = self.0.lock().unwrap();
-        Ok(inner_ref.ota_metadata.clone().unwrap_or_default())
+        inner_ref
+            .ota_metadata
+            .clone()
+            .ok_or(RAMStorageError::NotFound)
     }
     fn reset_ota_metadata(&self) -> Result<(), Self::Error> {
         let _ = self.0.lock().unwrap().ota_metadata.take();
         Ok(())
     }
 }
+#[cfg(feature = "ota")]
+impl<Iterable, Storage: OtaMetadataStorage> OtaMetadataStorage for Iterable
+where
+    for<'a> &'a Iterable: IntoIterator<Item = &'a Storage>,
```

Comment:
will only work on collection that implement  `impl<'a, T,> IntoIterator for &'a  ...`

Reply:
I have one small concern here. I don't have a fully worked counterexample, but I wonder a bit about using something like a HashSet as the type backing IntoIterator, because the iteration order isn't deterministic.